qiskit_experiments/framework/experiment_data.py

```diff
@@ -12,93 +12,1803 @@
+
+    def copy(self, new_name: Optional[str] = None):
+        """Creates a copy of the figure data"""
+        name = new_name or self.name
+        return FigureData(figure=self.figure, name=name, metadata=copy.deepcopy(self.metadata))
+
+    def __json_encode__(self) -> Dict[str, Any]:
+        """Return the json representation of the figure data"""
+        return {"figure": self.figure, "name": self.name, "metadata": self.metadata}
+
+    @classmethod
+    def __json_decode__(cls, args: Dict[str, Any]) -> "FigureData":
+        """Initialize a figure data from the json representation"""
+        return cls(**args)
+
+    def _repr_png_(self):
+        if isinstance(self.figure, MatplotlibFigure):
+            b = io.BytesIO()
+            self.figure.savefig(b, format="png", bbox_inches="tight")
+            png = b.getvalue()
+            return png
+        else:
+            return None
+
+    def _repr_svg_(self):
+        if isinstance(self.figure, str):
+            return self.figure
+        if isinstance(self.figure, bytes):
+            return str(self.figure)
+        return None
+
+
+class ExperimentData:
+    """Qiskit Experiments Data container class.
+
+    This class handles the following:
+    1. Storing the data related to an experiment - the experiment's metadata,
+       the analysis results and the figures
+    2. Manaing jobs and adding data from jobs automatically
+    3. Saving/Loading data from the result database
+    """
+
+    _metadata_version = 1
+    _job_executor = futures.ThreadPoolExecutor()
+
+    _json_encoder = ExperimentEncoder
+    _json_decoder = ExperimentDecoder
+
+    _metadata_filename = "metadata.json"
+
+    def __init__(
+        self,
+        experiment: Optional["BaseExperiment"] = None,
+        backend: Optional[Backend] = None,
+        service: Optional[IBMExperimentService] = None,
+        parent_id: Optional[str] = None,
+        job_ids: Optional[List[str]] = None,
+        child_data: Optional[List[ExperimentData]] = None,
+        verbose: Optional[bool] = True,
+        db_data: Optional[ExperimentDataclass] = None,
+        **kwargs,
+    ):
+        """Initialize experiment data.
+
+        Args:
+            experiment: Optional, experiment object that generated the data.
+            backend: Optional, Backend the experiment runs on.
+            service: The service that stores the experiment results to the database
+            parent_id: Optional, ID of the parent experiment data
+                in the setting of a composite experiment
+            job_ids: Optional, IDs of jobs submitted for the experiment.
+            child_data: Optional, list of child experiment data.
+            verbose: Optional, whether to print messages
+            db_data: Optional, a prepared ExperimentDataclass of the experiment info;
+            overrides other db parameters.
+        """
+        if experiment is not None:
+            backend = backend or experiment.backend
```

Comment:
`backend` takes priority over `experiment.backend` since it is passed explicitly. I don't want to limit the users is case the experiment itself does not tell the full story (although in the current flow, the backend is indeed taken from the experiment).

Reply:
Maybe a least inform about the override?

Reply:
I'm not sure why. In case `backend` was passed by mistake? It is not passed by default. I added a comment to the parameter documentation.